docs/articles/documentation/using-testcafe/common-concepts/screenshots-and-videos.md

```diff
@@ -0,0 +1,308 @@
+
+### Basic Video Options
+
+TestCafe supports the following video options:
+
+Option | Type | Description | Default Value
+------ | ---- | ----------- | --------------
+`failedOnly` | Boolean | `true` to record only failed tests; `false` to record all tests. | `false`
+`singleFile` | Boolean | `true` to save the entire recording as a single file; `false` to create a separate file for each test. | `false`
+`ffmpegPath` | String | The path to the FFmpeg codec executable. | Auto-detected
+`pathPattern` | String | A pattern that defines how TestCafe composes the relative path to a video file and the file name. See [Where Screenshots and Videos Are Saved](#where-screenshots-and-videos-are-saved). | See [Default Path Patterns](#default-path-patterns).
+
+You can specify video options in either of the following ways:
+
+* the [--video-options](../command-line-interface.md#--video-options-optionvalueoption2value2) command line flag,
+
+    ```sh
+    testcafe chrome test.js --video artifacts/videos --video-options singleFile=true,failedOnly=true,pathPattern=${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4
+    ```
+
+* the `options` parameter of the [runner.video](../programming-interface/runner.md#video) API method,
+
+    ```js
+    runner.video('artifacts/videos', {
+        singleFile: true,
+        failedOnly: true,
+        pathPattern: '${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4'
+    });
+    ```
+
+* the [videoOptions](../configuration-file.md#videooptions) configuration file property.
+
+    ```json
+    {
+        "videoOptions": {
+            "singleFile": true,
+            "failedOnly": true,
+            "pathPattern": "${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4"
+        }
+    }
+    ```
+
+### Video Encoding Options
+
+You can specify all the options supported by the FFmpeg library. Refer to [the FFmpeg documentation](https://ffmpeg.org/ffmpeg.html#Options) for information about the available options.
+
+To provide video encoding options, use either of the following:
+
+* the [--video-encoding-options](../command-line-interface.md#--video-encoding-options-optionvalueoption2value2) command line flag,
+
+    ```sh
+    testcafe chrome test.js --video artifacts/videos --video-encoding-options r=20,aspect=4:3
+    ```
+
+* the `encodingOptions` parameter of the [runner.video](../programming-interface/runner.md#video) API method,
+
+    ```js
+    runner.video('artifacts/videos', { }, {
+        r: 20,
+        aspect: '4:3'
+    });
+    ```
+
+* the [videoEncodingOptions](../configuration-file.md#videoencodingoptions) configuration file property.
+
+    ```json
+    {
+        "videoEncodingOptions": {
+            "r": 20,
+            "aspect": "4:3"
+        }
+    }
+    ```
+
+## Where Screenshots and Videos Are Saved
+
+You specify the base path to the directory that stores screenshots or videos when you turn these features on. See [Enable Screenshots](#enable-screenshots) and [Enable Video Recording](#enable-video-recording).
```

Comment:
You **can** specify the base path to the directory that stores screenshots or videos when you turn these features on. See [Enable Screenshots](#enable-screenshots) and [Enable Video Recording](#enable-video-recording).

Reply:
By saying 'you specify', I mean 'you normally specify'. Or, we can be more explicit, because actually you cannot enable screenshots without specifying the base path. So, we can say:

*You should specify the base path to the directory that stores screenshots or videos to enable these features.*

Reply:
I agree